docs/reference/fields.txt

```diff
@@ -447,6 +447,81 @@ matches strings containing "hello" before a newline, besides strings ending in
 This is because the meaning of ``$`` is different between PCRE and Ruby
 regular expressions.
 
+
+.. _typed-arrays:
+
+Field Type: Typed Arrays
+------------------------
+
+Mongoid supports declaring a typed array field on models. This field will
+enforce the type of the elements in the array by ``mongoizing`` each element
+inserted into the array. To make a typed array field:
+
+.. code:: ruby
+
+  class Band
+    include Mongoid::Document
+    field :members, type: Array(String)
```

Comment:
The `Array(String)` syntax does not extend to Set or Hash. If you are intending to support type restrictions for the other container types  I think it is worth figuring out a syntax that would work for all of them.

Reply:
I'm not necessarily considering supporting type restrictions for other container types right now. 

It seems silly to make a Mongoid::StringIntegerHash class for every Type x Type combination. 

Do you have any recommendations on what an alternate syntax could be?

Reply:
Using square brackets `[ ]` instead of parenthesis `( )` happens to work:

```ruby
Array[String]  #=> [String]
Set[String]    #=> #<Set: {String}>
Hash[String, Integer]  #=> {String=>Integer}
```

I also think we should consider if there can be an equivalent Symbol syntax for each of these.

```ruby
:string_array
:string_set
:string_to_integer_hash
```

where `*_array`, `*_set`, and `*_to_*_hash` are privileged symbols